Branch-SDK/src/io/branch/referral/ServerRequestInitSession.java

```diff
@@ -71,6 +71,26 @@ public boolean handleBranchViewIfAvailable(ServerResponse resp) {
         return isBranchViewShowing;
     }
 
+    @Override
+    public void onRequestSucceeded(ServerResponse response, Branch branch) {
+        // Check for any Third party SDK for data handling
+        String appOrigin = "";
+        try {
+            if (response.getObject().has(Defines.Jsonkey.AppOrigin.getKey())) {
+                appOrigin = response.getObject().getString(Defines.Jsonkey.AppOrigin.getKey());
+            }
+            boolean isFabricEnabled = appOrigin.equals(Defines.Jsonkey.AppOriginTwitterFabric.getKey());
```

Comment:
Hey Steve - Just for the sake of not surprising any existing customers by messing up their Answers dash, we decided to only enable the data integration for users who added Branch through Fabric. So, to answer your question, App Origin is set when the app is created or updated via Fabric. 

Looking forward, this value is something that we can easily expose as an integration to our partners via the dash, but it would be opt in for existing Branch folks.